renku/core/management/repository.py

```diff
@@ -540,7 +559,36 @@ def import_from_template(self, template_path, metadata, force=False):
                 rendered_content = template.render(metadata)
                 destination = Path(Template(str(destination)).render(metadata))
                 destination.write_text(rendered_content)
+                checksums[str(rel_path)] = self._content_hash(destination)
             except IsADirectoryError:
                 destination.mkdir(parents=True, exist_ok=True)
             except TypeError:
                 shutil.copy(file, destination)
+
+        self.template_checksums.parent.mkdir(parents=True, exist_ok=True)
+
+        with open(self.template_checksums, "w") as checksum_file:
+            json.dump(checksums, checksum_file)
+
+    def check_immutable_template_files(self, *paths):
+        """Check paths and return a list of those that are marked immutable in the project template."""
+        if not self.project.immutable_template_files:
+            return []
+
+        immutable_paths = []
+
+        for path in paths:
+            relative_path = Path(path)
+
+            if str(relative_path) in self.project.immutable_template_files:
+                immutable_paths.append(path)
+
+        return immutable_paths
```

Comment:
This would be a shorter version of the same function:
```
immutable_template_files = self.project.immutable_template_files or []
return [p for p in paths if p in immutable_template_files]
```